pkg/daemon/writer.go

```diff
@@ -43,9 +43,27 @@ func NewNodeStateStatusWriter(c snclientset.Interface, n string, f func()) *Node
 func (writer *NodeStateStatusWriter) Run(stop <-chan struct{}, refresh <-chan Message, syncCh chan<- struct{}, destDir string, runonce bool, platformType utils.PlatformType) {
 	glog.V(0).Infof("Run(): start writer")
 	msg := Message{}
+
+	var (
+		metaData    *utils.OSPMetaData
+		networkData *utils.OSPNetworkData
+		err         error
+	)
+
+	if platformType == utils.VirtualOpenStack {
+		metaData, err = utils.ReadOpenstackMetaData()
+		if err != nil {
+			glog.Errorf("Run(): failed to read OpenStack meta_data: %v", err)
+		}
+		networkData, err = utils.ReadOpenstackNetworkData()
```

Comment:
Instead of adding extra parameters to `pollNicStatus`, shall we move `metaData` and `networkData` to struct `NodeStateStatusWriter`?

Reply:
done